tests/coverage/coverage.cpp

```diff
@@ -61,10 +61,76 @@ TEST(Coverage, importWithNonHrefXlink)
         "  </connection>\n"
         "</model>\n";
 
-    // Parse
     libcellml::ParserPtr parser = libcellml::Parser::create();
-    parser->parseModel(e);
-    EXPECT_EQ(size_t(0), parser->issueCount());
+    parser->parseModel(in);
+    EXPECT_EQ(size_t(1), parser->issueCount());
+}
+
+TEST(Coverage, importWithNamespaceViolations)
+{
+    const std::string in =
+        "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+        "<model name=\"\" xmlns=\"http://www.cellml.org/cellml/2.0#\">\n"
+        "  <import xmlns:xlink=\"http://www.w3.org/1999/xlinked\" xlink:href=\"some-other-model.xml\">\n"
+        "    <component component_ref=\"component_in_that_model\" name=\"component_in_this_model\"/>\n"
+        "    <!-- Comment in an import block -->\n"
+        "  </import>\n"
+        "  <import xmlns:xlink=\"http://www.w3.org/1999/xlink\" xmlns=\"http://www.cellml.org/notcellml/2.0#\" xlink:href=\"some-other-model.xml\">\n"
+        "    <component component_ref=\"component_in_that_model\" name=\"component_in_this_model\"/>\n"
+        "    <!-- Comment in an import block -->\n"
+        "  </import>\n"
+        "  <component name=\"component_bob\">\n"
+        "    <variable name=\"variable_bob\" units=\"dimensionless\"/>\n"
+        "  </component>\n"
+        "  <connection component_2=\"component_in_this_model\" component_1=\"component_bob\">\n"
+        "    <map_variables variable_2=\"variable_import\" variable_1=\"variable_bob\"/>\n"
+        "  </connection>\n"
+        "</model>\n";
+
+    libcellml::ParserPtr parser = libcellml::Parser::create();
+    parser->parseModel(in);
+    EXPECT_EQ(size_t(5), parser->issueCount());
```

Comment:
Would be nice to check the exact issues. Would make it easier to appreciate what is being tested here, as well as for code maintenance.

Reply:
These are coverage tests so we are not interested in the very specifics here like we are for other tests.

Reply:
Should this really be a coverage test though?

Reply:
The namespace violations is part of the main testing this is cover for branches that are unlikely to be hit. Even hard to hit branches because you have to weave a certain path.